File: PathoFroh/src/main/java/com/patho/main/model/interfaces/DataList.java
package com.patho.main.model.interfaces;

import java.util.Set;

import javax.persistence.Transient;

import com.patho.main.model.PDFContainer;
import com.patho.main.template.PrintDocument.DocumentType;

/**
 * Classes have a data list
 * 
 * @author andi
 *
 */
public interface DataList extends ID {

	public Set<PDFContainer> getAttachedPdfs();

	public void setAttachedPdfs(Set<PDFContainer> attachedPdfs);

	public String getPublicName();

	public default boolean contaisReportType(DocumentType type) {
		if (getAttachedPdfs() != null) {
			return getAttachedPdfs().stream().anyMatch(p -> p.getType().equals(type));
		}

		return false;
	}

	/**
	 * Adds a pdf to a datalist
	 * 
	 * @param pdfTemplate
	 */
	public default void addReport(PDFContainer pdfTemplate) {
		getAttachedPdfs().add(pdfTemplate);
	}

	/**
	 * Removes a report
	 * 
	 * @param pdfTemplate
	 */
	public default void removeReport(PDFContainer pdfTemplate) {
		getAttachedPdfs().remove(pdfTemplate);
	}

	/**
	 * Returns true if the pdf is in the datalist
	 * 
	 * @param pdfTemplate
	 * @return
	 */
	public default boolean containsReport(PDFContainer pdfTemplate) {
		return getAttachedPdfs() != null && getAttachedPdfs().contains(pdfTemplate);
	}
}

File: PathoFroh/src/main/kotlin/com/patho/main/model/interfaces/DataList.kt
package com.patho.main.model.interfaces

import com.patho.main.config.PathoConfig
import com.patho.main.model.PDFContainer
import com.patho.main.template.PrintDocumentType
import java.io.File
import javax.persistence.Transient

interface DataList : ID, PatientAccessible {
    var attachedPdfs: MutableSet<PDFContainer>

    val publicName: String;

    open val fileRepositoryBase: File
        @Transient
        get() = File(PathoConfig.FileSettings.FILE_REPOSITORY_PATH_TOKEN + patient?.id)

    open fun containsReportType(type: PrintDocumentType): Boolean {
        return attachedPdfs.any { p -> p.type == type }
    }

    open fun addReport(pdfTemplate: PDFContainer) {
        attachedPdfs.add(pdfTemplate)
    }

    open fun removeReport(pdfTemplate: PDFContainer) {
        attachedPdfs.remove(pdfTemplate)
    }

    open fun containsReport(pdfTemplate: PDFContainer): Boolean {
        return attachedPdfs.contains(pdfTemplate)
    }
}